src/runtime_src/core/common/info_aie.cpp

```diff
@@ -0,0 +1,717 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#define XRT_CORE_COMMON_SOURCE
+#include "info_aie.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+#include <boost/algorithm/string.hpp>
+#include <boost/optional/optional.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+namespace qr = xrt_core::query;
+using ptree_type = boost::property_tree::ptree;
+
+// major, minor and patch version of the schema of AIE metadata.
+const uint32_t sc_major = 1;
+const uint32_t sc_minor = 0;
+const uint32_t sc_patch = 0;
+
+namespace {
+// Convert graph status from integer to a human readable
+// string format for better understanding.
+inline std::string
+graph_status_to_string(int status) {
+  switch(status)
+  {
+    case 0:	return "stop";
+    case 1:	return "reset";
+    case 2:	return "running";
+    case 3:	return "suspend";
+    case 4:	return "end";
+    default:	return "idle"; // Consider graph status as "idle" for all the default cases
```

Comment:
We will leave it and the PR is approved.  But this is a 'duh' comment and maybe I misunderstood what you said in your comment to my comment? I thought you meant that the graph status retrieved from the device itself specifies that the value for idle is anything but 0-4 so that this switch statement is forced to use `default:` for idle.  Your comment makes it sounds as if it is your choice.  Never mind.

Reply:
The graph status is retrieved from aie driver, not from zocl driver. Hence it's unknow what are the possible default status values. It's upto us how we call the default status. And as per the discussion, it is better to use "unknown" than "idle" . 
I have updated the code. Please let me know if this will make it more clear.